onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -842,6 +840,16 @@ void addGlobalMethods(py::module& m, const Environment& env) {
   m.def(
       "disable_telemetry_events", []() -> void { platform_env.GetTelemetryProvider().DisableTelemetryEvents(); },
       "Disables platform-specific telemetry collection.");
+  m.def(
+      "create_and_register_allocator", [&env](const OrtMemoryInfo& mem_info, const OrtArenaCfg& arena_cfg) -> void {
+        // TODO: Remove the const qualifier for the environment instance in the function parameter ?
+        // It strictly needn't be const and would save the casting away the const in this method
+        // CreateAndRegisterAlloctor() method mutates the environment and is hence a non-const method
+        auto st = const_cast<Environment&>(env).CreateAndRegisterAlloctor(mem_info, &arena_cfg);
```

Comment:
Can we do this as part of addObjectMethods which doesn't take const Environment?

Reply:
It seems a bit out of place to add a global method away from a section dedicated to it. I removed the 'const' qualifier for the Environment instance (and strictly it doesn't have to be 'const' since we have run into a valid case where we might have to invoke a non-const method)